Cards/CardBase.gd

```diff
@@ -31,29 +31,49 @@ func parse_card_data(card_data: Dictionary) -> void:
 	# TODO
 	pass
 
-func _apply_all_effects(target: Entity) -> void:
-	for effect_data: EffectData in card_effects_data:
-		effect_data.apply_effect_data(target)
+func _apply_all_effects(targets: Array[Entity], effects: Array[EffectData]) -> void:
+	for target : Entity in targets:
+		for effect : EffectData in effects:
+			effect.apply_effect_data(target)
 
 func can_play_card(caster: Entity, target: Entity) -> bool:
 	return caster.get_party_component().can_play_on_entity(application_type, target)
 
 
 func on_card_play(caster: Entity, targets: Array[Entity]) -> void:
 	#Split up targeted attacks and all attacks
+	var target_effects: Array[EffectData] = get_target_effects()
+	var all_effects: Array[EffectData] = get_all_effects()
+	
+	#Apply single target
+	_apply_all_effects(targets, target_effects)
+	#Get every unit that is to be affected by card
+	var all_target : Array[Entity] = get_all_targets()
+
+	#apply effect to every target
+	_apply_all_effects(all_target, all_effects)
```

Comment:
doesn't this assume all the effects have the same targets ? this might not be the case for a card like "does 1 damage to a selected enemy, heal 1 to self"

Reply:
Right now, they do all have the same target, and it's kinda difficult to affect both sides with one card. There should definitaly be some changes to this, but maybe it should happen with the target changes

Reply:
did you check #42 in link with this ?

Reply:
Yeah, I saw, I was just thinking that fixing this issue would be better to do in #42, but I could do a bit of refactoring to enable the changes more

Reply:
thank you

Reply:
I can't actually think of a good way of fixing this without just refactoring the whole thing, so I think I'll leave it to #42

Reply:
alright fair enough, this might be a bit out of the scope anyway